src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -76,17 +76,23 @@ def select_interpreter_for_targets(self, targets):
     """Pick an interpreter compatible with all the specified targets."""
     tgts_with_compatibilities = []
     filters = set()
-    for target in targets:
-      if isinstance(target, PythonTarget) and target.compatibility:
-        tgts_with_compatibilities.append(target)
-        filters.update(target.compatibility)
+
+    if self._python_setup.get_options().is_flagged('interpreter_constraints'):
+      # Reverse to prioritize flagged option over default value
+      filters.update(reversed(self._python_setup.get_options().interpreter_constraints))
```

Comment:
We want to get the last constraint in the list here (the CLI-supplied constraint) via:
`filters.add(self._python_setup.get_options().interpreter_constraints[-1])`
We need to add the element here so the set object does not update the set with each character of the constraint string.